x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Chunk.java

```diff
@@ -130,28 +112,41 @@ protected TypeResolution resolveType() {
         if (childrenResolved() == false) {
             return new TypeResolution("Unresolved children");
         }
-        return isString(field(), sourceText(), FIRST).and(Options.resolve(options, source(), SECOND, ALLOWED_OPTIONS, this::verifyOptions));
-    }
 
-    private void verifyOptions(Map<String, Object> optionsMap) {
-        if (options == null) {
-            return;
+        TypeResolution fieldResolution = isString(field(), sourceText(), FIRST);
+        if (fieldResolution.unresolved()) {
+            return fieldResolution;
         }
 
-        Integer numChunks = (Integer) optionsMap.get(NUM_CHUNKS);
-        if (numChunks != null && numChunks < 0) {
-            throw new InvalidArgumentException("[{}] cannot be negative, found [{}]", NUM_CHUNKS, numChunks);
+        return chunkingSettings == null ? TypeResolution.TYPE_RESOLVED : validateChunkingSettings();
```

Comment:
we can rewrite this to:
```
return isString(field(), sourceText(), FIRST).and(this::validateChunkingSettings);
```

and then in `validateChunkingSettings`, we just need to add a check right at the beginning:
```
if (options== null) {
     return TypeResolution.TYPE_RESOLVED;
}
```